src/Core/Services/IOrganizationService.cs

```diff
@@ -55,7 +55,7 @@ public interface IOrganizationService
     Task<List<Tuple<OrganizationUser, string>>> DeleteUsersAsync(Guid organizationId,
         IEnumerable<Guid> organizationUserIds, Guid? deletingUserId);
     Task UpdateUserGroupsAsync(OrganizationUser organizationUser, IEnumerable<Guid> groupIds, Guid? loggedInUserId);
-    Task UpdateUserResetPasswordEnrollmentAsync(Guid organizationId, Guid userId, string resetPasswordKey, Guid? callingUserId);
+    Task UpdateUserResetPasswordEnrollmentAsync(Guid organizationId, Guid userId, string resetPasswordKey, IUserService userService, Guid? callingUserId);
```

Comment:
Please do not send services through method parameters, if this would cause a circular dependency, then we need to move this method into a command.

Reply:
@justindbaur 
I was just following what appeared to be a convention in this file:
https://github.com/bitwarden/server/blob/d020c49c0e3957e4bf08a38be32f306e4225fdda/src/Core/Services/IOrganizationService.cs#L43-L44

I assumed that this was because services on the same layer were not allowed to depend on each other directly. Now that you mention it though, there does seem to be a circular dependency. Guess that means I will have to refactor the entire `UpdateUserResetPasswordEnrollmentAsync` into a `Command` 👍

Reply:
I gave it a fair shot spending half a day converting both `UpdateUserResetPasswordEnrollmentAsync` and `AcceptUserAsync` into commands but ran into even more circular dependencies because `UserService.SetPasswordAsync` depends on `AcceptUserAsync` and `AcceptUserAsync` depends on `UserService` meaning that if I convert `SetPasswordAsync` into a command I will end up with a dependency between two commands instead. Which in turn would require that we un-nest that spaghetti by moving the inter-command dependency into controllers and who knows what issues that might cause.

I've removed `IUserService` from `UpdateUserResetPasswordEnrollmentAsync` and instead call `AcceptUserAsync` directly from the controller. I still need to pass in `IUserService` but that method has always needed that service and so I'm at least not making it worse.

I think this whole circular dependency mess needs to be handled in a separate tech debt story

Reply:
If it has already happened in this file and it's to much work for now then your solution is fine. But yes, lets make a tech debt story for it. As far as I'm aware the only other service we pass along in parameters is `ICurrentContext` and I also dislike that one 😆.